shapes/Square.cpp

```diff
@@ -1,8 +1,11 @@
 #include "Square.hpp"
 #include <iostream>
 
-Square::Square(Color c)
-    : Rectangle(c) {}
+Square::Square(Color color)
+    : Rectangle(color) {}
+
+Square::Square(double x)
+    : Rectangle(x, x, Color::Green) {}
```

Comment:
ok, but this one is not a delegation. You simply call a base class constructors here. Constructor delegation takes place when you call other constructor of your own class.